django_fsm/__init__.py

```diff
@@ -403,7 +403,11 @@ def _collect_transitions(self, *args, **kwargs):
         def is_field_transition_method(attr):
             return (inspect.ismethod(attr) or inspect.isfunction(attr)) \
                 and hasattr(attr, '_django_fsm') \
-                and attr._django_fsm.field in [self, self.name]
+                and (attr._django_fsm.field in [self, self.name] or \
+                     (issubclass(self.model, attr._django_fsm.field.model) and\
+                      attr._django_fsm.field.name == self.name and \
+                      attr._django_fsm.field.creation_counter == self.creation_counter))
```

Comment:
@ticosax Can you explain why `creation_counter` is needed and why it is only needed for 3.2 and not for previous versions?

Reply:
because creation counter is part of the __hash__ function of the field. 
https://github.com/django/django/blob/231de683d86374c2b74da2185efc6ddfb5eb3341/django/db/models/fields/__init__.py#L543
Which means, `creation_counter` is probably the way to go to avoid collisions between 2 fields that are foreign to each other.

This assumption is based on the diff that introduced the regression for django-fsm https://github.com/django/django/commit/502e75f9ed5476ffe8229109acf0c23999d4b533